### DISCUSSION TITLE
✔ Feature Request: Support window resize of the main app window

### DISCUSSION THREAD

**Discussion by ForrestFeng** [IndiePython/nodezator#15] [Feature Request]
Have tried the nodezator app and I like it very much. I have a big screen but the main widows' size is small and fixed. It would be great if the window can be resized.

I have tried to update the constants to match my screen size with no luck(My system is Windows 10, screen size is 1920 x 1080, also tried to set the text scale to 100%, 120% or 150% does not work either).

pygameconstants.py
```
SCREEN_WIDTH, SCREEN_HEIGHT = 1280, 720
FLAG, DEPTH = 0, 32
```

![image](https://user-images.githubusercontent.com/5875631/175530184-8e7fb1a1-8a02-4581-a96b-9901b2bccc83.png)

**Comment by ForrestFeng** [comment]
**Update for main window size.** 

Tried to call set_mode without args, I got a full-screen app window that looks much better. 

pygameconstants.py
```
#SCREEN = set_mode((SCREEN_WIDTH, SCREEN_HEIGHT), FLAG, DEPTH)
SCREEN = set_mode()
```
BUT, there is one bug in full-screen mode, the open file dialog layout is broken. Though it is broken, it still functions, and I can open the file want. 

![image](https://user-images.githubusercontent.com/5875631/175533063-1dde641c-4c2b-4ef7-be48-3184070b2704.png)

**Comment by KennedyRichard** [comment]
Hello, Mr. Feng,

This is the most requested feature right now and I'm committed to implementing it as soon as possible. I've even commented on it recently here: https://github.com/KennedyRichard/nodezator/discussions/6#discussioncomment-2997839

I'm glad you took the initiative to try your own solution, your insight is really helpful.

The only reason I didn't make the window resizable from the beginning was because I wanted to ship Nodezator asap and focus on fixing critical bugs, and making the window resizable would require some weeks between adjusting the interface to reposition correctly when the screen is resized and testing everything.

As I said in the linked discussion, I'll open a new discussion where I'll list the next features to be implemented (including making the window resizable), gather feedback from the community and start working right away. I'll be doing that today or  tomorrow. For now I'm just committing and merging the changes to fix issue #10 <img src="https://media3.giphy.com/media/eCO8PynsAKrfRR2XIX/giphy.gif"/>

**Comment by KennedyRichard** [comment]
Mr. Feng, I took the liberty to convert your Feature Request issue into a discussion, since it doesn't describe an issue per se (a bug or something that causes problems/warnings, etc.).

The discussion I mentioned in my last message above was already published and the feature you and a lot of users have been requesting (resizable window) is already listed and scheduled there to start being implemented on June 28th. Here's a link to the discussion: #14

**Comment by KennedyRichard** [comment]
Forgot to close this discussion after implementing the feature back on July 2022. Closing it now!

Peace
